genai-perf/genai_perf/parser.py

```diff
@@ -351,17 +351,29 @@ def parse_goodput(values):
 
 
 def _infer_prompt_source(args: argparse.Namespace) -> argparse.Namespace:
-
     args.synthetic_input_files = None
+    args.payload_input_file = None
 
     if args.input_file:
+        input_file_str = str(args.input_file)
         if str(args.input_file).startswith("synthetic:"):
```

Comment:
You may want to reuse the arg above (input_file_str). 

Also, this function is starting to get long. I'm thinking we might want to refactor parser sometime soon (CC: @debermudez @nv-hwoo). This function is starting to feel long. I don't think we need to slow down development by breaking this into its own class with functions yet, but I do think we should take a look at how we can restructure parser.py as a whole and break it into a cleanly-organized directory.